### ISSUE TITLE
highway=cycleway not visible on zoom=12

### ISSUE THREAD

**Issue by sonora** [osmandapp/Osmand#2849]
Reminder to investigate: Getting complaints that highway=cycleway are not visible in "Touring view" map style on zoom 12 anymore. No obvious flaw in our code at first sight, also seem to be correctly included in rendering_types.xml.

**Comment by vshcherb** [comment]
Controversial, though indeed it is correctly set in the types. obf2osm might clarify situation

**Comment by xmd5a2** [comment]
It is defined at line 8822 in default.render.xml
```xml
					<case minzoom="12" tag="highway" value="cycleway">
						<case appMode="bicycle">
							<apply_if moreDetailed="true" maxzoom="13" strokeWidth_5="0.7" pathEffect_5="4_2"/>
							<apply minzoom="14" strokeWidth_4="1.4:1.4" strokeWidth_5="1:1" pathEffect_5="$cyclewayPathEffect"/>
						</case>
						<case minzoom="14" strokeWidth_4="1.3:1.3" strokeWidth_5="$cyclewayStrokeWidth" pathEffect_5="$cyclewayPathEffect"/>
						<apply color_5="$cyclewayColor"/>
						<apply_if nightMode="true" color="#77002a2a">
							<apply_if minzoom="13" maxzoom="13" strokeWidth_4="0.6"/>
							<apply_if minzoom="14" maxzoom="14" strokeWidth_4="1:1"/>
							<apply_if minzoom="15" strokeWidth_4="1.2:1.2"/>
						</apply_if>
					</case>
```

**Comment by sonora** [comment]
Yes, but shouldn't line 822 ff in Touring-view overwrite that?
```
			<group>
				<filter tag="highway" value="cycleway"/>
				<groupFilter color_5="$cyclewayColorDay" pathEffect_5="4_2" color="$null">
					<filter appMode="bicycle" minzoom="14" strokeWidth_5="2"/>
					<filter appMode="pedestrian" minzoom="14" strokeWidth_5="2"/>
					<filter minzoom="12" maxzoom="12" strokeWidth_5="0.5"/>
					<filter minzoom="13" strokeWidth_5="1"/>
				</groupFilter>
			</group>
```

**Comment by vshcherb** [comment]
Hard to say this group syntax is quite confusing.

**Comment by xmd5a2** [comment]
@sonora Do you really want to display cycleways at zoom 12? There are almost indistinguishable from the strokes of the road. This will look like this https://dl.dropboxusercontent.com/u/4624786/osm/Screenshots/Screenshot%20from%202016-07-28%2000%3A33%3A43.png
Anyway, I've done a test and can say that zoom limiter in default.render.xml is here
```xml
<case minzoom="13" tag="highway" value="cycleway">
```
Looks like style inheritance works different with old <group> <filter> syntax. This can be a problem.

**Comment by sonora** [comment]
Yes, I knew there should be some inheritance issue outside our xml code - thanks for investigating!! Not sure when it started.

Leonid: I agree that in "crowded" areas having all travel options on zoom 12 is overkill. But my clientele for the Touring view relies on zoom 12 showing a **complete map with all line (travel)-objects.**

Many live in areas with EXTREMELY thin highway density, often much less than 1 per 10 square  miles, and they bought this issue up. (Actually in a desert area, where an old railroad corridor was apparently converted to a "cyclepath", not even sure where it is, likely in California).

I wonder if we ever get to tackling my old idea of having dynamic minzooms, where we would add (inherit) features from more detailed (higher) zoom levels onto maps screens with lower zoom level for areas where few objects are mapped. (Essentially: Adding more objects to low-density areas, the opposite of removing overlapping detail from "overcrowded" areas).

**Comment by ligfietser** [comment]
Please render cycleways a lot sooner than it is now (zoom 15!?)
In the default general osmand map, cycleways render at a very high zoom level, even after paths!

See these images here: https://osm.org/go/0E6zjPuk--?m=
Cycleroutes, cycling nodes and unclassified roads are rendered at this level, bikepaths not:
![01](https://user-images.githubusercontent.com/9638726/73608775-26cd0400-45c7-11ea-9757-dd705c73060a.jpg)

Next level, tracks are rendered:
![02](https://user-images.githubusercontent.com/9638726/73608776-27659a80-45c7-11ea-9df2-5c5372174135.jpg)

Next zoom, paths are rendered:
![03](https://user-images.githubusercontent.com/9638726/73608777-27659a80-45c7-11ea-8f43-a47b0afd1d44.jpg)

Cycleways are rendered:
![04](https://user-images.githubusercontent.com/9638726/73608778-27659a80-45c7-11ea-8aee-5fb7e56c2306.jpg)

Better is to render them at the same level as image 1 (cycling routes)

**Comment by ligfietser** [comment]
I think the issue has to do something with this line:
`<case baseAppMode="default" tag="cycleway" value="" moreDetailed="false" maxzoom="15" order="-1"/>`

If I lower this maxzoom in the defaul.render.xml (eg "11"), cycleways in my Openfietsmap style are rendered way more earlier. Don't know how to implement this in my own rendering style though.
